adapter/integration-test/src/adapter-datasource-trigger.test.js

```diff
@@ -0,0 +1,263 @@
+      datasourceId: datasourceId,
+      dataLocation: `/data/${id}`
+    })
+  })
+
+  test('POST datasources/{id}/trigger dynamic defaultvalues', async () => {
+    const datasourceResponse = await request(URL)
+      .post('/datasources')
+      .send(dynamicDatasourceConfig)
+    const datasourceId = datasourceResponse.body.id
+
+    const dataMetaData = await request(URL)
+      .post(`/datasources/${datasourceId}/trigger`)
+      .send(null)
+
+    const id = dataMetaData.body.id
+    const data = await request(URL)
+      .get(`/data/${id}`)
+      .send()
+
+    const delResponse = await request(URL)
+      .delete('/datasources/')
+      .send()
+
+    expect(delResponse.status).toEqual(204)
+    expect(datasourceResponse.status).toEqual(201)
+    expect(dataMetaData.status).toEqual(200)
+    expect(dataMetaData.body.id).toBeGreaterThanOrEqual(0)
+    expect(dataMetaData.body.location).toEqual(`/data/${id}`)
+    expect(data.status).toEqual(200)
+    expect(data.body.id).toBe(dynamicDatasourceConfig.protocol.parameters.defaultParameters.id)
+
+    // check for rabbitmq notification
+    expect(publishedEvents.get(EXECUTION_SUCCESS_TOPIC)).toContainEqual({
+      datasourceId: datasourceId,
+      dataLocation: `/data/${id}`
+    })
+  })
+
+  test('POST datasources/{id}/trigger static', async () => {
+    const datasourceResponse = await request(URL)
+      .post('/datasources')
+      .send(staticDatasourceConfig)
+    const datasourceId = datasourceResponse.body.id
+
+    const dataMetaData = await request(URL)
+      .post(`/datasources/${datasourceId}/trigger`)
+      .send(null)
+
+    const id = dataMetaData.body.id
+    const normalData = await request(URL)
+      .get(`/data/${id}`)
+      .send()
+
+    const delResponse = await request(URL)
+      .delete('/datasources/')
+      .send()
+
+    expect(delResponse.status).toEqual(204)
+    expect(dataMetaData.status).toEqual(200)
+    expect(dataMetaData.body.id).toBeGreaterThanOrEqual(0)
+    expect(dataMetaData.body.location).toEqual(`/data/${id}`)
+    expect(normalData.status).toEqual(200)
+    expect(normalData.body.id).toBe('1')
+
+    // check for rabbitmq notification
+    expect(publishedEvents.get(EXECUTION_SUCCESS_TOPIC)).toContainEqual({
+      datasourceId: datasourceId,
+      dataLocation: `/data/${id}`
+    })
+  })
+
+  test('POST datasource/{id}/trigger FAIL', async () => {
+    const brokenDatasourceConfig = Object.assign({}, staticDatasourceConfig)
+    brokenDatasourceConfig.protocol.parameters.location = 'LOL'
+    const datasourceResponse = await request(URL)
+      .post('/datasources')
+      .send(brokenDatasourceConfig)
+
+    const datasourceId = datasourceResponse.body.id
+
+    const triggerResponse = await request(URL)
+      .post(`/datasources/${datasourceId}/trigger`)
+      .send()
+
+    expect(triggerResponse.status).toBeGreaterThan(300) // request should fail (no 2xx status)
```

Comment:
why not make concrete? either 400 for bad request, or imho better 404 (because resource does not exist)

Reply:
I chose to leave it open because the validation in the adapter service does not work now and needs refactoring. It currently returns 404 no matter what error occurs, see #156